static/js/public/snap-details/channelMap.js

```diff
@@ -1,294 +1,425 @@
-      let fallbackRisk;
-      for (let j = 0; j < i; j++) {
-        const channelName = track === LATEST ? risks[j] : `${track}/${risks[j]}`;
-        if (channelMap[channelName]) {
-          fallbackRisk = risks[j];
-        }
-      }
+    // initialize architecture select
+    const archSelect = document.querySelector('[data-js="arch-select"]');
+
+    archSelect.innerHTML = architectures.map(arch => `<option value="${arch}">${arch}</option>`).join('');
+
+    this.arch = this.channelMapData['amd64'] ? 'amd64' : architectures[0];
+
+    // capture events
+    this.bindEvents();
+  }
 
-      if (fallbackRisk) {
-        versionEl.innerHTML = `No release in ${risk} channel, using ${fallbackRisk} release.`;
+  sortRows(rows){
+    // split tracks into strings and numbers
+    let numberTracks = [];
+    let stringTracks = [];
+    let latestTracks = [];
+    rows.forEach(row => {
+      // numbers are defined by any string starting any of the following patterns:
+      //   just a number – 1,2,3,4,
+      //   numbers on the left in a pattern – 2018.3 , 1.1, 1.1.23 ...
+      //   or numbers on the left with strings at the end – 1.1-hotfix
+      if (row[0] === this.defaultTrack) {
+        latestTracks.push(row);
+      } else if (isNaN(parseInt(row[0].substr(0, 1)))) {
+        stringTracks.push(row);
       } else {
-        versionEl.innerHTML = `No release in ${risk} channel.`;
-        channelEl.classList.add('p-channel-map__row--closed');
+        numberTracks.push(row);
       }
-    }
-  });
-}
+    });
 
+    // Ignore case
+    stringTracks.sort(function (a, b) {
+      return a[0].toLowerCase().localeCompare(b[0].toLowerCase());
+    });
 
-function getArchTrackChannels(arch, track, channelMapData) {
-  const channels = channelMapData[arch][track];
-  const archTrackChannels = {};
+    stringTracks = latestTracks.concat(stringTracks);
 
-  channels.forEach(channel => {
-    archTrackChannels[channel.channel] = channel;
-  });
+    // Sort numbers (that are actually strings)
+    numberTracks.sort((a, b) => {
+      return b[0].localeCompare(a[0], undefined, { numeric: true, sensitivity: 'base' });
+    });
 
-  return archTrackChannels;
-}
+    // Join the arrays together again
 
-function setArchitecture(arch, packageName, channelMapData) {
-  if (!arch) {
-    return;
+    return stringTracks.concat(numberTracks);
   }
 
-  const tracks = Object.keys(channelMapData[arch]);
-
-  // split tracks into strings and numbers
-  let numberTracks = [];
-  let stringTracks = [];
-  tracks.forEach(track => {
-    // numbers are defined by any string starting any of the following patterns:
-    //   just a number – 1,2,3,4,
-    //   numbers on the left in a pattern – 2018.3 , 1.1, 1.1.23 ...
-    //   or numbers on the left with strings at the end – 1.1-hotfix
-    if (isNaN(parseInt(track.substr(0, 1)))) {
-      stringTracks.push(track);
-    } else {
-      numberTracks.push(track);
+  bindEvents() {
+    globalEvents.addEvents({
```

Comment:
Events module quite nicely cleans up the events code here, I'm just not sure if we need all of these events handled globally on `body`? Event delegation has obvious benefits (especially when HTML is created dynamically and events don't need to be recreated), but it would be good to have control over it - for example to handle them on the root element of channel map instead of the body.

Reply:
I've made the events module a class so we can define a default element to bind to - though in the case of the channelMap I'm just letting it bind to the default body still